src/ansys/speos/core/bsdf.py

```diff
@@ -0,0 +1,339 @@
+
+    @property
+    def reflection(self) -> bool:
+        """Contains the BSDF Reflection data."""
+        return self._reflection
+
+    @reflection.setter
+    def reflection(self, value: bool):
+        if value:
+            self._reflection = value
+        else:
+            self._reflection = value
+            self._brdf = None
+
+    @property
+    def brdf(self) -> Collection[BxdfDatapoint]:
+        """List of BRDFDatapoints."""
+        return self._brdf
+
+    @brdf.setter
+    def brdf(self, value: Collection[BxdfDatapoint]):
+        check = any([not bxdf.type for bxdf in value])
+        if check is True or value is None:
+            self._brdf = value
+        else:
+            msg = "One or multiple datapoints are transmission datapoints"
+            raise ValueError(msg)
+
+    @property
+    def btdf(self) -> Collection[BxdfDatapoint]:
+        """List of BTDFDatapoints."""
+        return self._btdf
+
+    @btdf.setter
+    def btdf(self, value: Collection[BxdfDatapoint]):
+        check = any([bxdf.type for bxdf in value])
+        if check is True or value is None:
+            self._btdf = value
+        else:
+            msg = "One or multiple datapoints are reflection datapoints"
+            raise ValueError(msg)
+
+    @property
+    def nb_incidents(self) -> list[int]:
+        """Number of incidence angle for reflection and transmission.
+
+        Returns
+        -------
+        list[int]:
+            first value of the list is nb of reflective data, second value is nb of transmittive
+            data
+        """
+        return [len(self.brdf), len(self.btdf)]
+
+    @property
+    def incident_angles(self) -> list[Union[list[float], None], Union[list[float], None]]:
+        """List of incident angles for reflection and transmission.
+
+        Returns
+        -------
+        list[Union[list[float], None],Union[list[float], None]]
+            Returns a nested list of incidence angels for reflective and transmittive
+            data if not available the value will be None
+        """
+        r_angle = [bxdf.incident_angle for bxdf in self.brdf]
+        t_angle = [bxdf.incident_angle for bxdf in self.btdf]
+        return [r_angle, t_angle]
+
+    def reset(self):
+        """Reset BSDF data to what was stored in file."""
+        pass
+
+
+class BxdfDatapoint:
+    """Class to store a BxDF data point.
+
+    Parameters
+    ----------
+    bxdf_type : bool
+        true for transmittive date, False for reflective
+    incident_angle : float
+        incident angle in radian
+    theta_values : Collection[float]
+        list of theta values for the bxdf data matrix, in radian
+    phi_values : Collection[float]
+        list of phi values for the bxdf data matrix, in radian
+    bxdf : Collection[float]
+        nested list of bxdf values in 1/sr
+    anisotropy : float
+        Anisotropy angle in radian
+    """
+
+    def __init__(
+        self,
+        bxdf_type: bool,
```

Comment:
```suggestion
        is_brdf: bool,
```
maybe change to that, so the user can xxx.is_brdf is true/false to check the type.

Reply:
good idea